### DISCUSSION TITLE
Search String Issues (New User!)

### DISCUSSION THREAD

**Discussion by Idiotic-Virtue** [Lailloken/Exile-UI#413] [Q&A] [answered]
So have installed and imported my exile levelling and have that working perfectly (is great!).

Trying to have a single regex for all vendors. Have set this up (probably wrong!) but whenever i try it does not work. When in the UI and trying the test it says callibration required.

When setting up the omni key does not seem to recognise any of my mouse keys so have had to set to "J" on the keyboard.

Can some explain why i am clearly an idiot and how to solve?
![Screenshot 2024-07-24 213016](https://github.com/user-attachments/assets/fa7f6f25-2078-4dee-9440-e73968f05c21)
![Screenshot 2024-07-24 213034](https://github.com/user-attachments/assets/8af02951-5fec-4334-a340-7fff5687f86c)
![Screenshot 2024-07-24 213144](https://github.com/user-attachments/assets/d1b80287-801e-41a0-9d20-fee00482243c)

**Comment by Lailloken** [comment, marked as answer]
> When setting up the omni key does not seem to recognise any of my mouse keys so have had to set to "J" on the keyboard.

- You have to be careful because it may activate stuff when you're typing. The tool doesn't know whether or not you're chatting, so it will activate every single time you press "J".

- If you leave the edit-field blank, it's on middle-mouse by default (you can still use it in game to use spells, etc.)

- If you want to use-side buttons, they're usually called "xbutton1" and "xbutton2", so you'll have to type that in and check which button is which on your mouse
<br>

**Okay, so here's how the search-strings feature works:**
- you set up a use-case

  - you have already done that, but it's too specific: Hargan doesn't have a unique vendor window (it's looks the same as the other NPCs' windows)

  - so remove that and use the "vendor" you have also created

- you have to calibrate the use-case (basically, you tell the tool what the window looks like), so the tool knows which menu to show

  - click "cal" to open a snipping tool

  - use the snipping tool to take a screenshot of a **`unique part`** of the vendor window, e.g.:

    ![image](https://github.com/user-attachments/assets/4cd16bde-47f5-4d4f-80cf-c042a89398eb)

  - click "test" to see if the tool can find this part of the window

- you add "entries" to the use-case

  - basically, what you want to see on this menu:

  ![image](https://github.com/user-attachments/assets/95fba51b-5c47-42c9-a7ce-db7d32088dd5)

  - click "edit", and in the new window add entries by typing distinct names

  - so in your case, probably something like "Hargan gear"

- on the right, you can add one or more strings to each entry

  - if you add one, that will simply be pasted when selecting "Hargan gear" from the context-menu

  - if you add multiple (one per line), it will paste the first string and enter "scrolling mode" where you can scroll the mouse-wheel to switch back and forth between the individual strings

- when set up correctly, pressing the omni-key while the vendor window is on screen should activate the context-menu and let you pick strings
<br>

If you need more details/help, check [this wiki-entry](https://github.com/Lailloken/Lailloken-UI/wiki/Search-strings) or simply reply here

**Reply by Idiotic-Virtue** [reply]
Thanks so much - yup that snip thing was what i was missing! Thanks so much :)
